calico_node/startup/startup.go

```diff
@@ -267,7 +274,11 @@ func configureIPsAndSubnets(node *api.Node) {
 		adm := os.Getenv("IP_AUTODETECTION_METHOD")
 		cidr := autoDetectCIDR(adm, 4)
 		if cidr != nil {
-			// We autodetected an IPv4 address so update the value in the node.
+			// We autodetected an IPv4 address, if we have no previous IP, or it is different than the previous IP,
```

Comment:
Sorry I didn't notice before, but this check is not quite sufficient as it only covers the auto-detection case.

I think we also need to handle the case when the IP address has been explicitly specified in the environment.